84050 сделал :

bookinisto opened the pull request:
- добавил в UserService метод getByIdUsers , а так же в котролллер соответсвующий эндпоинт
- написал тесты на этот метод в контроллер и сервис

bookinisto committed:
84050 сделал :
- добавил в UserService метод getByIdUsers , а так же в котролллер соответсвующий эндпоинт
- написал тесты на этот метод в контроллер и сервис

hhzktoeto commented:
Надо устранить конфликты при слиянии, у тебя они явно видны в pull request'е
<img width="658" height="178" alt="image" src="https://github.com/user-attachments/assets/943833e6-d202-4317-8565-2f22919b9b02" />

bookinisto committed:
84050 сделал :
- добавил в UserService метод getByIdUsers , а так же в котролллер соответсвующий эндпоинт
- написал тесты на этот метод в контроллер и сервис

bookinisto committed:
84050 сделал :
- добавил в UserService метод getByIdUsers , а так же в котролллер соответсвующий эндпоинт
- написал тесты на этот метод в контроллер и сервис

RinatBeybutov commented on src/main/java/school/faang/user_service/service/user/UserServiceImpl.java:
```diff
@@ -139,4 +137,9 @@ public void linkChatId(Long userId, Long chatId) {
+
+    @Override
+    public List<UserDto> getByIdUsers(List<Long> usersId) {
+        return usersId.stream().map(userId -> userMapper.toUserDto(userRepository.getByIdOrThrow(userId))).toList();
```
каждая операция стрима с новой строчки, а также нужно использовать userRepository.findAllById(usersId)

RinatBeybutov commented on src/main/java/school/faang/user_service/service/user/UserServiceImpl.java:
```diff
@@ -53,10 +53,8 @@ public UserDto create(UserCreateDto userDto) {
-                .preference(PreferredContact.fromString(userDto.contact()))
-                .build();
+        ContactPreference pref = ContactPreference
+                .builder().user(user).preference(PreferredContact.fromString(userDto.contact())).build();
```
также с новой строчки каждая операция билдера

RinatBeybutov commented on src/main/resources/application.yaml:
```diff
@@ -55,8 +55,10 @@ cloud:
       endpoint: https://storage.yandexcloud.net
-      access-key: ${AWS_ACCESS_KEY:}
-      secret-key: ${AWS_SECRET_KEY:}
+      access-key: asmndbh79asd
```
здесь нужно написать вот так:
${AWS_ACCESS_KEY:asmndbh79asd}

Ниже аналогично

RinatBeybutov commented on src/test/java/school/faang/user_service/controller/user/UserControllerTest.java:
```diff
@@ -46,6 +48,15 @@ class UserControllerTest {
+    private long userId = 1L;
+    private String userName = "Bob";
+    private String email = "@mail";
+    private List<Long> userIds = List.of(userId);
```
давай это добро вынесем в класс с тестовыми данными

RinatBeybutov commented on src/test/java/school/faang/user_service/service/user/UserServiceImplTest.java:
```diff
@@ -52,6 +55,17 @@ class UserServiceImplTest {
+    private List<Long> userIds = List.of(userId);
+    private String phone = "88899212";
+    private String aboutMe = "Fine";
+    private String avatarUrl = "www/www";
```
здесь так же в тестовый класс

RinatBeybutov commented:
и еще здесь нужно поправить соответственно запросу в post service(там уже есть запрос на всех пользователей в feign через post и тело запроса

bookinisto committed:
84050 сделал :
 - добавил в UserRepository метод
 List<User> findAllByIdIn(List<Long> userIds)
 - переписал реализацию метода getByIdUsers в UserServiceImpl
 -  исправил    access-key: ${AWS_ACCESS_KEY:asmndbh79asd}
        secret-key: ${AWS_SECRET_KEY:asmndbh79asd}
 - удалил из UserController <List<UserDto>> getListUsers
 - из UserControllerTest удалил тест метод на getListUsers
 -

bookinisto committed:
84050 сделал :
 - добавил в UserRepository метод
 List<User> findAllByIdIn(List<Long> userIds)
 - переписал реализацию метода getByIdUsers в UserServiceImpl
 -  исправил    access-key: ${AWS_ACCESS_KEY:asmndbh79asd}
        secret-key: ${AWS_SECRET_KEY:asmndbh79asd}
 - удалил из UserController <List<UserDto>> getListUsers
 - из UserControllerTest удалил тест метод на getListUsers
 -
